girder_worker/docker/transform.py

```diff
@@ -0,0 +1,136 @@
+import sys
+
+from girder_worker_utils.transform import Transform
+
+
+class StdOut(Transform):
+    def transform(self):
+        from girder_worker.docker.io import (
+            StdStreamWriter
+        )
+        return StdStreamWriter(sys.stdout)
+
+
+class StdErr(Transform):
+    def transform(self):
+        from girder_worker.docker.io import (
+            StdStreamWriter
+        )
+        return StdStreamWriter(sys.stderr)
+
+
+class ContainerStdOut(Transform):
+
+    def transform(self):
+        return self
+
+    def open(self):
+        # noop
+        pass
+
+
+class ContainerStdErr(Transform):
+
+    def transform(self):
+        return self
+
+    def open(self):
+        # noop
+        pass
+
+
+class ProgressPipe(Transform):
+
+    def __init__(self, path):
+        super(ProgressPipe, self).__init__()
+        self._path = path
+
+    def transform(self, task):
+        from girder_worker import utils
+        from girder_worker.docker.io import (
+            NamedPipe,
+            NamedPipeReader,
+            ReadStreamConnector
+        )
+        super(ProgressPipe, self).transform()
+        job_mgr = task.job_manager
```

Comment:
job_manager may exist but be None. Not sure what the 'right' thing to do in that case is,  but wanted to make you aware